### PR TITLE
Fix storage drivers dropping non EOF errors when listing repositories

### PR DESCRIPTION
**Pull request by hinshun**
This fixes errors other than io.EOF from being dropped when a storage driver
lists repositories. For example, filesystem driver may point to a missing
directory and errors, which then gets subsequently dropped.
    
Signed-off-by: Edgar Lee <edgar.lee@docker.com>

### CODE REVIEW

File: registry/storage/catalog.go [hunk 1]
```diff
@@ -25,12 +25,12 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 		return 0, errors.New("no space in slice")
 	}
 
-	root, err := pathFor(repositoriesRootPathSpec{})
-	if err != nil {
-		return 0, err
+	root, errVal := pathFor(repositoriesRootPathSpec{})
```
Comment (RichardScothern) [inline registry/storage/catalog.go]: is the name change of this variable necessary?

Comment (RichardScothern) [inline registry/storage/catalog.go]: ah, just expanded upwards and realized it's the named return value 😭

Comment (dmcgowan) [inline registry/storage/catalog.go]: I think I would prefer just changing the named return to `err` than make the change here

Comment (hinshun) [inline registry/storage/catalog.go]: Yeah, I think it makes sense to keep it consistent with how we name it err in other places.

**Comment by RichardScothern** [comment]
LGTM

**Comment by dmcgowan** [comment]
LGTM

**Pull request merged**